Reference: Security overview (#377)
appuio/openshift4-docs: docs/modules/ROOT/pages/references/security.adoc

= Security

== Access Management

=== Long Lived Tokens

The default OpenShift 10 year emergency admin token is disabled and replaced by a short lived token.

See xref:oc4:ROOT:references/architecture/emergency_credentials.adoc[].

=== Two Factor Authentication

> VSHN managed OpenShift clusters are configured to use two factor authentication through the VSHN SSO for the web console by default.

Reviewer: ```suggestion
VSHN managed OpenShift clusters are configured to use two factor authentication through the https://kb.vshn.ch/kb/vshn-login.html[VSHN Login] for the web console by default.
```

> The configuration is centrally managed according to best practices.

Reviewer: which configuration is manage to best practice? the VSHN Login page?

Author: Specified OpenID client configuration

> See xref:oc4:ROOT:references/architecture/single_sign_on.html[].

Reviewer: ```suggestion
See xref:oc4:ROOT:references/architecture/single_sign_on.adoc[].
```

=== Sudo

> VSHN managed OpenShift clusters are configured to deny dangerous operations by default.

Reviewer: ```suggestion
VSHN Managed OpenShift clusters are configured to deny dangerous operations by default.
```

See xref:oc4:ROOT:explanations/sudo.adoc[].

== Containers

> === Priviledged Containers and Build Strategies

Reviewer 2: ```suggestion
=== Privileged Containers and Build Strategies
```

> VSHN managed OpenShift clusters are configured to deny the use of privileged containers and build strategies by default.

Reviewer: ```suggestion
VSHN Managed OpenShift clusters are configured to deny the use of privileged containers and build strategies by default.
```

I think the Brand is Managed OpenShift, so capital M

See xref:oc4:ROOT:explanations/pod_security.adoc[].

=== Seccomp

> VSHN managed OpenShift clusters are configured to use the CRI-O seccomp profile `runtime/default` by default.

Reviewer: ```suggestion
VSHN Managed OpenShift clusters are configured to use the CRI-O seccomp profile `runtime/default` by default.
```

This restricts some syscalls that could shorten exploit chains.

> 

Reviewer 2: probably makes sense to mention SCC & PSA

```suggestion

=== OpenShift SecurityContextConstraints

VSHN managed OpenShift clusters run workloads with OpenShift's `restricted-v2` SCC by default.

[quote,'https://docs.openshift.com/container-platform/4.16/authentication/managing-security-context-constraints.html#default-sccs_configuring-internal-oauth[from the OpenShift default SCC documentation]']
--
The `restricted-v2` SCC:

* Ensures that pods cannot run as privileged
* Ensures that pods cannot mount host directory volumes
* Requires that a pod is run as a user in a pre-allocated range of UIDs
* Requires that a pod is run with a pre-allocated MCS label
* Requires that a pod is run with a preallocated FSGroup
* Allows pods to use any supplemental group
* ALL capabilities are dropped from containers.
* The `NET_BIND_SERVICE` capability can be added explicitly.
* `seccompProfile` is set to `runtime/default` by default.
* `allowPrivilegeEscalation` must be unset or set to `false` in security contexts
--

=== Kubernetes Pod Security Admission

VSHN managed OpenShift clusters are configured to validate workloads against the `restricted` Kubernetes Pod Security Standard by default.
Currently, the Pod Security Admission is configured to only warn users about violating configurations.

See xref:explanations/pod_security.adoc[] for details on how PSA and OpenShift SCCs interact.
```

== Network

=== Load Balancer

> VSHN managed OpenShift clusters include, depending on the cloud provider, hardened load balancers based on HAProxy.

Reviewer: ```suggestion
VSHN Managed OpenShift clusters include, depending on the cloud provider, hardened load balancers based on HAProxy.
```

Addictional firewalls and jumphosts can be configured on request.

=== Cilium

> VSHN managed OpenShift clusters uses the hardened enterprise version of Cilium as the default network plugin.

Reviewer: Should we also link to https://products.vshn.ch/openshift/cilium.html or is that more sales documentation?

https://cilium.io/[Cilium]

=== Default Namespace Isolation

> VSHN managed OpenShift clusters are configured to deny traffic between different namespaces by default.

Reviewer: ```suggestion
VSHN Managed OpenShift clusters are configured to deny traffic between different namespaces by default.
```

== Updates

> All VSHN managed OpenShift clusters and their load balancers are automatically updated to the latest version of OpenShift and the latest security patches.

Reviewer: ```suggestion
All VSHN Managed OpenShift clusters and their load balancers are automatically updated to the latest version of OpenShift and the latest security patches.
```
ditto above

If not otherwise agreed, the updates are applied weekly during the communicated maintenance window.

See xref:oc4:ROOT:references/architecture/upgrade_controller.adoc[].

== Backup

> VSHN managed OpenShift clusters include backups of all Kubernetes manifests and the raw etcd data to a secure location by default.

Reviewer: ```suggestion
VSHN Managed OpenShift clusters include backups of all Kubernetes manifests and the raw etcd data to a secure location by default.
```

> 

Reviewer: ```suggestion

See:
* xref:oc4:ROOT:references/explanations/disaster_recovery.adoc[].
* xref:oc4:ROOT:references/how-tos/recover-from-backup.adoc[].
```

== Inventory

All VSHN managed configuration and software for the OpenShift clusters are stored in a central Git repository.
Software versions are reported in a central repository and actively monitored.

We use https://syn.tools/syn/index.html[Project Syn] to manage the inventory.

=== Staged Rollouts

New software versions are rolled out in a staged manner to ensure that the software is stable and secure.

== Monitoring and Logging

=== Capacity Monitoring

> VSHN managed OpenShift clusters include capacity monitoring by default.

Reviewer: ```suggestion
VSHN Managed OpenShift clusters include capacity monitoring by default.
```

VSHN gets notified if the cluster is running out of resources.

> 

Reviewer: ```suggestion

See xref:oc4:ROOT:explanations/cluster_monitoring.adoc[]
```

=== Logging

> VSHN managed OpenShift clusters include logging by default.

Reviewer: ```suggestion
VSHN Managed OpenShift clusters include logging by default.
```

A copy of the Kubernetes audit logs is stored in a secure location at VSHN.

> 

Reviewer: ```suggestion

See also https://docs.appuio.cloud/user/tutorials/logging/query-logs.html[Query application logs]
```

== VSHN

pass:[<!-- vale off -->]
pass:[<!-- direct quotes -->]

> > We are convinced that transparency and certified processes improve data security and confidentiality.
> > We are ISO 27001 certified and work according to the strict FINMA guidelines to ensure the security and confidentiality of client data at all times.
> 
> > VSHN is the first Kubernetes Certified Service Provider (KCSP) in Switzerland, Red Hat Advanced CCSP Partner and we are ISO 27001 certified (you can download and view our ISO certificate), we work according to the strict FINMA guidelines and are ISAE 3402 Report Type 2 audited.

Reviewer: ```suggestion
[quote]
____
We are convinced that transparency and certified processes improve data security and confidentiality.
We are ISO 27001 certified and work according to the strict FINMA guidelines to ensure the security and confidentiality of client data at all times.

VSHN is the first Kubernetes Certified Service Provider (KCSP) in Switzerland, Red Hat Advanced CCSP Partner and we are ISO 27001 certified (you can download and view our ISO certificate), we work according to the strict FINMA guidelines and are ISAE 3402 Report Type 2 audited.
____

```

pass:[<!-- vale on -->]

* https://www.vshn.ch/en/about/awards-certifications/[VSHN Awards & Certifications]
* https://handbook.vshn.ch/ism_policies.html[Information Security Management Policies]
* https://kb.vshn.ch/kb/security_vulnerability_process.html[Security and Vulnerability Handling Process]

> 

Reviewer: ```suggestion
== Deletion
See https://kb.vshn.ch/oc4/how-tos/cloudscale/decommission.html
```

Perhaps one sentence about cluster deletion? Something that make it clear we will get rid of data if not used any more.
